uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/bookingrequest/BookingRequestActivity.kt

```diff
@@ -1,97 +1,100 @@
 
     private var holdOpenForPaymentFlow = false
 
-    private var presenter: BookingRequestMVP.Presenter = BookingRequestPresenter(this,
-            KarhooUISDK.analytics,
-            KarhooPreferenceStore.getInstance(context.applicationContext),
-            KarhooApi.tripService,
-            KarhooApi.userStore)
+    private lateinit var presenter: BookingRequestContract.Presenter
 
     private val bookingComments: String
         get() = bookingRequestCommentsWidget.getBookingOptionalInfo()
 
-    init {
-        View.inflate(context, R.layout.uisdk_booking_request, this)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        presenter = BookingRequestActivityPresenter(this,
+                KarhooUISDK.analytics,
+                KarhooPreferenceStore.getInstance(this.applicationContext),
+                KarhooApi.tripService,
+                KarhooApi.userStore)
+
         isGuest = isGuest()
         attachListeners()
-        bookingRequestFlightDetailsWidget.setHintText(context.getString(R.string.kh_uisdk_add_flight_details))
+        bookingRequestFlightDetailsWidget.setHintText(this.getString(R.string.kh_uisdk_add_flight_details))
         presenter.setBookingFields(bookingRequestPassengerDetailsWidget.allFieldsValid())
         TextViewCompat.setTextAppearance(bookingRequestLabel, R.style.ButtonText)
+
+        extras?.let { extras ->
+            val quote = extras.getParcelable<Quote>(BOOKING_REQUEST_QUOTE_KEY)
+            val bookingStatus = extras.getParcelable<BookingStatus>(BOOKING_REQUEST_STATUS_KEY)
+
+            if (quote != null) {
+                bookingRequestButton.onLoadingComplete()
+                bookingRequestLayout.visibility = View.VISIBLE
+                presenter.setBookingStatus(bookingStatus)
+                presenter.showBookingRequest(
+                        quote = quote,
+                        outboundTripId = extras.getString(BOOKING_REQUEST_OUTBOUND_TRIP_ID_KEY),
+                        bookingMetadata = extras.getSerializable(BOOKING_REQUEST_METADATA_KEY) as HashMap<String, String>?
+                )
+                //initialiseChangeCard(quote)
+            }
+        }
+
     }
 
+    override fun handleExtras() {}
+
     override fun showGuestBookingFields(details: PassengerDetails) {
-        val constraintSet = ConstraintSet()
-        constraintSet.constrainHeight(R.id.bookingRequestScrollView, 0)
-        constraintSet.clone(bookingRequestLayout)
-        constraintSet.connect(R.id.bookingRequestScrollView, ConstraintSet.TOP, ConstraintSet.PARENT_ID,
-                ConstraintSet.TOP, 0)
-        constraintSet.applyTo(bookingRequestLayout)
-
-        bookingRequestPassengerDetailsWidget.visibility = VISIBLE
-        bookingRequestCommentsWidget.visibility = VISIBLE
-        passengerDetailsHeading.visibility = VISIBLE
+        bookingRequestPassengerDetailsWidget.visibility = ConstraintLayout.VISIBLE
```

Comment:
maybe we should import `import android.view.View.VISIBLE` and write like this:
`bookingRequestPassengerDetailsWidget.visibility = VISIBLE`
It would be more visible and we don't need to go thru the ConstraintLayout class